models/update_checker.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"code.gitea.io/gitea/modules/proxy"
+	"code.gitea.io/gitea/modules/setting"
+
+	"github.com/hashicorp/go-version"
+)
+
+// GiteaUpdateChecker returns error when new version of Gitea is available
```

Comment:
Do our definitions of `error` differ, or are you required by the caller to only return an error?
I mean, yes, I can understand it at least to some extent, but I definitely would like it more if **the best case would not be an error case**.
I can understand printing it to console and logs, but I do not know if treating this case as an error is the best solution for this…

Reply:
The interface for cron actions is such that returning an error is the only viable path to have a message in the system notices. Otherwise the message would be purely "cron completed successfully". 

I had an idea where I created a new table and added the most recent JSON fetched to a row in that table as a place to store then in admin dashboard fetch that row and display notice if there is an update, and if enabled email admins (max 1 time per update).